src/core/lib/promise/party.h

```diff
@@ -82,6 +82,51 @@ class Party : public Activity, private Wakeable {
   Party(const Party&) = delete;
   Party& operator=(const Party&) = delete;
 
+  // When calling into a Party from outside the promises system we often would
+  // like to perform more than one action.
+  // This class tries to acquire the party lock just once - if it succeeds then
+  // it runs the party in its destructor, effectively holding all wakeups of the
+  // party until it goes out of scope.
+  // If it fails, presumably some other thread holds the lock - and in this case
+  // we don't attempt to do any buffering.
+  class WakeupHold {
+   public:
+    WakeupHold() = default;
+    explicit WakeupHold(Party* party)
+        : prev_state_(party->state_.load(std::memory_order_relaxed)) {
+      // Try to lock
```

Comment:
Should there be a check on prev_state_ to check if the kLocked bit is set before we perform the compare_exchange_weak operation ?